src/Management/src/Endpoint/ActuatorEndpointMapper.cs

```diff
@@ -35,30 +33,34 @@ public IEndpointConventionBuilder Map(IEndpointRouteBuilder endpointRouteBuilder
         var collection = new HashSet<string>();
         conventionBuilder ??= new ActuatorConventionBuilder();
 
-        foreach (IContextName name in _contextNames)
+        foreach (EndpointContexts context in Enum.GetValues<EndpointContexts>())
```

Comment:
Yikes! thanks for pointing this out. I am thinking the flags enum is not a good match and I will solve this with a regular enum and adding a collection of enum to management options.

Reply:
The flags enum on `ManagementEndpointOptions` was replaced with an `IList<>` of enum constants, which is not correct because that allows duplicates. An `ISet<>` would be better. Then we conceptually have a set of constants... which is precisely what a `[Flags]` enum represents!

I still believe a flags enum is the better choice, because it's more expressive and maintainable (which matters, now that it's part of the public API surface). When a user configures "All", and we add another entry in the future, it will automatically pick that up. Alternatively, a user could configure "Actuator,CloudFoundry" to explicitly ignore any future additions.

It is also more performant and uses less memory, although that may not be much of a concern here.

Validation is simpler as well. Instead of iterating the entries in the ISet, it's simply a matter of:
```c#
if ((value | EndpointContexts.All) != EndpointContexts.All)
{
    throw new Exception("Invalid combination of flags");
}
```
which would throw if the user configured "Actuator,123".

I'd be happy to help in getting this implemented correctly. Let's discuss this at our meeting today.